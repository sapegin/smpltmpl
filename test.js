'use strict';

const template = require('./index').template;
const templateFromFile = require('./index').templateFromFile;

const isNode4 = process.version.startsWith('v4.');

describe('template()', () => {
	it('should apply template to a file', () => {
		const result = template('Hello, ${foo}!', { foo: 'Bar' });
		expect(result).toBe('Hello, Bar!');
	});

	it('should escape `', () => {
		const fn = () => template('Hello, \\`${foo}\\`!', { foo: 'Bar' });
		expect(fn).not.toThrowError();

		const result = fn();
		expect(result).toBe('Hello, `Bar`!');
	});

	it('should throw if template has a syntax error', () => {
		const fn = () => template('Hello\n${foo!', { foo: 'Bar' });
		// Node 4 in some case doesnâ€™t have position information
		expect(fn).toThrowError(
			isNode4 ? 'Error in template untitled' : 'Error in template untitled:2'
		);
	});

	it('should throw if context variable is missed', () => {
		const fn = () => template('Hello\nworld of templates ${foo}!', { bar: 'Bar' });
		expect(fn).toThrowError('Error in template untitled:2:22\nfoo is not defined');
	});

	it('should show a given file name in an error message', () => {
		const fn = () => template('Hello\n${foo}!', {}, 'pizza.txt');
		expect(fn).toThrowError('Error in template pizza.txt:2:3');
	});

	it('should a code frame', () => {
		const fn = () => template('Hello\nworld ${foo}!', { bar: 'Bar' });
		expect(fn).toThrowErrorMatchingSnapshot();
	});
});

describe('templateFromFile()', () => {
	it('should apply template to a file', () => {
		const result = templateFromFile('test/template.txt', { foo: 'Bar' });
		expect(result).toBe('Hello, Bar!');
	});

	it('should throw if a template file not found', () => {
		const fn = () => templateFromFile('pizza', {});
		expect(fn).toThrowError('Template file not found: pizza');
	});

	it('should throw in case of other error while reading a file', () => {
		const fn = () => templateFromFile('test', {});
		expect(fn).toThrowError('EISDIR');
	});
});
