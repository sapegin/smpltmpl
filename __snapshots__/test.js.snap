// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template() should a code frame 1`] = `
"Error in template untitled:2:9
foo is not defined

  1 | Hello
  2 | world \${foo}!"
`;
